content/rsf-spec/index.md

```diff
@@ -0,0 +1,263 @@
+* Registers specification: [`CANONREP`][canon-rep] (canonical representation).
+  Note that, in turn, it depends on [RFC8259](https://tools.ietf.org/html/rfc8259).
+
+```abnf
+log              = command *(CRLF command) [CRLF]
+command          = add-item / append-entry / assert-root-hash
+
+assert-root-hash = %s"assert-root-hash" HTAB hash
+
+add-item         = %s"add-item" HTAB CANONREP
+
+append-entry     = %s"append-entry" HTAB type HTAB key HTAB timestamp HTAB hash-list
+type             = "user" / "system"
+key              = alphanum / %x2D / %x5F
+hash-list        = hash *(list-separator hash)
+hash             = "sha-256:" 64(HEXDIG) ; sha-256
+list-separator   = ";" ; hash list separator
+
+alphanum         = ALPHA / DIGIT
+
+;                timestamp
+timestamp        = date "T" time
+date             = century year DSEP month DSEP day ; date YYYY-MM-DD
+time             = hour TSEP minute TSEP second TZ ; time HH:MM:SSZ
+
+;                date
+century          = 2DIGIT ; 00-99
+year             = 2DIGIT ; 00-99
+month            = 2DIGIT ; 01-12
+day              = 2DIGIT ; 01-28, 01-29, 01-30, 01-31 based on month/year
+DSEP             = "-"    ; date separator
+
+;                time
+hour             = 2DIGIT ; 00-24
+minute           = 2DIGIT ; 00-59
+second           = 2DIGIT ; 00-58, 00-59, 00-60 based on leap-second rules
+TSEP             = ":"    ; time separator
+TZ               = "Z"    ; timezone
+```
+
+### Media type
+
+The current media type is `application/uk-gov-rsf`. It should change to
+`application/vnd.rsf` to align with [RFC6838](https://tools.ietf.org/html/rfc6838).
+
+### REST API
+
+TODO
+
+The current implementation uses `GET /download-rsf`. The main issue with that
+is that diverges from the rest of the API where serialisation is expressed
+either via suffix or via media type. The problem with using the same approach,
+say `GET /register.rsf` is that we are not providing the same information when
+querying `GET /register.json`.
+
+What is a good name for a resource that represents the whole raw database?
+
+```
+# RSF
+
+GET /{db resource}.rsf
+
+GET /{db resource}
+Accept: application/vnd.rsf
+
+
+# JSON Lines — Hypothetical
+
+GET /{db resource}.jsonl
+
+GET /{db resource}.jsonl
+Accept: application/x-ndjson
+```
+
+
+### Commands
+
+#### <a id="assert-root-hash-command">`assert-root-hash` command</a>
+
+Asserts that the provided root hash is the same as the one computed from the
+current entry log as defined in the [Digital Proofs][digital-proofs]
+specification.
+
+##### Arguments
+
+1. The `hash` of the root of the tree.
+
+For example, the empty root hash:
+
+```
+assert-root-hash	sha-256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
+```
+
+#### <a id="add-item-command">`add-item` command</a>
+
+Adds a new [Item resource][item-res] to the register. It will require an
+[`append-entry` command](#append-entry-command) to make it visible to users.
```

Comment:
Instead of "make it visible to users", I would put "associate the item with a record key".

Reply:
umm, I like what you are suggesting but I don't like the idea of using the word "record". @michaelabenyohai ideas?

Reply:
How about "There must be a corresponding `append-entry` command that refers to the item's hash"?  Noting the later comment on "It is illegal to have orphan items", which insists on the `append-entry` for the file to be valid.

Reply:
That could work 👍